src/geode/model/mixin/core/blocks.cpp

```diff
@@ -126,11 +126,13 @@ namespace geode
                 }
             } );
         }
-        async::when_all( tasks.begin(), tasks.end() )
-            .then( [level] {
-                Logger::set_level( level );
-            } )
-            .wait();
+        auto all_tasks = async::when_all( tasks.begin(), tasks.end() );
```

Comment:
Why not when_all(…).get() here?

Reply:
If the when_all throws an exception, le Logger will not be reset to its initial logging level. It is unlikely (but possible) that the when_all throws but just to be safe.